arrow/src/main/java/org/apache/iceberg/arrow/vectorized/ArrowReader.java

```diff
@@ -268,10 +268,10 @@ private static final class VectorizedCombinedScanIterator implements CloseableIt
           .map(entry -> EncryptedFiles.encryptedInput(io.newInputFile(entry.getKey()), entry.getValue()));
 
       // decrypt with the batch call to avoid multiple RPCs to a key server, if possible
-      Iterable<InputFile> decryptedFiles = encryptionManager.decrypt(encrypted::iterator);
+      Iterator<InputFile> decryptedFiles = encryptionManager.decrypt(encrypted.iterator());
```

Comment:
While this isn't technically correct, I think it is better not to change a public API. If you think this is a significant problem, then we have to options:
1. Collect `encrypted` into a list
2. Create a `OneTimeIterable` to wrap `Stream::iterator` instead